### ISSUE TITLE
invalid function call panic

### ISSUE THREAD

**Issue by mvertes** [traefik/yaegi#230]
The following program `sample.go` triggers a panic:

```go
package main

import "time"

func main() {
    time.Date()
}
```

Expected result:
```console
$ go run ./sample.go
./sample.go:6:11 not enough arguments in call to time.Date
	have ()
	want (int, time.Month, int, int, int, int, int, *time.Location)
```

Got:
```console
$ yaegi ./sample.go
./sample.go:6:2: panic
panic: reflect: Call with too few input arguments [recovered]
	panic: reflect: Call with too few input arguments

goroutine 1 [running]:
github.com/containous/yaegi/interp.runCfg.func1(0xc0001720f0, 0xc0001dad00)
	/Users/marc/go/src/github.com/containous/yaegi/interp/run.go:107 +0x1e9
panic(0x17fb740, 0x1add260)
	/usr/local/Cellar/go/1.12.6/libexec/src/runtime/panic.go:522 +0x1b5
reflect.Value.call(0x1863340, 0x1947e40, 0x13, 0x1913388, 0x4, 0x20b8698, 0x0, 0x0, 0x1910a40, 0x190b201, ...)
	/usr/local/Cellar/go/1.12.6/libexec/src/reflect/value.go:364 +0x153b
reflect.Value.Call(0x1863340, 0x1947e40, 0x13, 0x20b8698, 0x0, 0x0, 0x2540008, 0x0, 0x49)
	/usr/local/Cellar/go/1.12.6/libexec/src/reflect/value.go:308 +0xa4
github.com/containous/yaegi/interp.callBin.func6(0xc0001720f0, 0xc0000684c0)
	/Users/marc/go/src/github.com/containous/yaegi/interp/run.go:837 +0x1a5
github.com/containous/yaegi/interp.runCfg(0xc0001dad00, 0xc0001720f0)
	/Users/marc/go/src/github.com/containous/yaegi/interp/run.go:112 +0x70
github.com/containous/yaegi/interp.(*Interpreter).run(0xc0000800c0, 0xc0001da500, 0xc000172050)
	/Users/marc/go/src/github.com/containous/yaegi/interp/run.go:93 +0x126
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc0000800c0, 0xc00002a140, 0x3a, 0x23a, 0xc00002a140, 0x3a, 0x0, 0x17fa600)
	/Users/marc/go/src/github.com/containous/yaegi/interp/interp.go:283 +0x32d
main.main()
	/Users/marc/go/src/github.com/containous/yaegi/yaegi.go:62 +0x443
```

**Comment by cosmos72** [comment]
This actually seems the correct behaviour: Go playground reports the same error while compiling
https://play.golang.org/p/DP4WH77kIvX
```
./prog.go:6:11: not enough arguments in call to time.Date
	have ()
	want (int, time.Month, int, int, int, int, int, *time.Location)
```

Maybe you meant something like:
```go
package main

import (
	"fmt"
	"time"
)

func main() {
	var t time.Time
	fmt.Println(t.Date())
}
```
which outputs
```
1 January 1
```
on both yaegi and Go playground?

**Comment by pborman** [comment]
Perhaps a similar issue:

```
package main

func foo() { }

func main() {
	foo() + "bob"
}
```

causes a panic:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xa8 pc=0x120924a]
```

I originally encountered this in interactive mode:

```
$ yaegi     
> import "fmt"
0xc000162150
> func foo() { fmt.Println("Hello World") }
0xc0001622c0
> foo() + " bob"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xa8 pc=0x120924a]
...
```

Clearly I missused foo() but I would have expected an error rather than a panic.

**Comment by bogen85** [comment]
This in the same issue as [issue #230](https://github.com/containous/yaegi/issues/280)

When the panic is being recovered, one gets this:
```
> import "fmt"
0xc00016a550
> func foo() { fmt.Println("Hello World") }
0xc0002163b0
> foo()
Hello World
0xc0002166d0
> foo()+" bob"
trapEval:  runtime error: invalid memory address or nil pointer dereference
> foo()
Hello World
0xc00016a9a0
> " bob "
 bob 
> 
```

**Comment by mvertes** [comment]
A bit of context. The processing of invalid code is bad today, as we have focused so far on running valid code first, so we could open this project and move forward.

We cumulate several issues where invalid code is not properly detected and trigger panic at execution instead of a clean error prior to execute.

We have syntax errors already detected by the Go AST parser, but ignored later by yaegi due to the  way we have integrated the REPL. I have a correction for that, i will submit it when I return from my current trip at gophercon). In some cases there is also a different behaviour wether the code is run in REPL mode vs batch mode. I want to address that as well.

Then there are semantic errors (syntax is valid, semantic not), to be caught during CFG. That requires some type check routines which are still missing.

Finally, there are legitimate panics occuring at runtime, that we want to propagate normally in batch mode as well as in REPL mode (So still poundering if a trapEval in REPL is necessary or not, provided that  we detect errors correctly at the right place, instead of executing bad code).
